tools/cli_commands/gpg_encrypt.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Callable, Iterable, Mapping, Optional
+
+from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils import config
+from reconcile import queries
+from reconcile.utils import gpg
+
+import json
+
+
+@dataclass
+class GPGEncryptCommandData:
+    vault_secret_path: str
+    vault_secret_version: int
+    secret_file_path: str
+    output: str
+    target_user: str
+
+
+class UserNotFoundException(Exception):
+    pass
+
+
+class ArgumentException(Exception):
+    pass
+
+
+class GPGEncryptCommand:
+    def __init__(
+        self,
+        users: Iterable[Mapping[str, str]],
+        secret_reader: SecretReader,
+        command_data: GPGEncryptCommandData,
+        gpg_encrypt_func: Callable,
+    ):
+        self._users = users
+        self._secret_reader = secret_reader
+        self._command_data = command_data
+        self._gpg_encrypt_func = gpg_encrypt_func
+
+    def _fetch_secret_content(self) -> str:
+        if self._command_data.vault_secret_path:
+            d = {"path": self._command_data.vault_secret_path}
+            if self._command_data.vault_secret_version > 0:
+                d["version"] = str(self._command_data.vault_secret_version)
+            secret = self._secret_reader.read_all(d)
+            return json.dumps(secret, sort_keys=True, indent=4)
+        elif self._command_data.secret_file_path:
+            with open(self._command_data.secret_file_path) as f:
+                return f.read()
+        else:
+            raise ArgumentException(
+                f"No argument given which defines how to fetch the secret {self._command_data}"
+            )
+
+    def _get_gpg_key(self) -> Optional[str]:
+        try:
+            return [
+                user.get("public_gpg_key")
+                for user in self._users
+                if user.get("org_username") == self._command_data.target_user
+            ][0]
```

Comment:
the org_username in the schema is unique, so maybe we could make it searchable. this would enable this code to only load the one user that is required.

https://github.com/app-sre/qontract-schemas/blob/86ba651bdc8b93880a5fed2466acc4cdf6aeeb0b/graphql-schemas/schema.yml#L1987

Reply:
This is a great idea Gerd! Will try that now, thanks! :bow:

Reply:
Fixed :+1: